src/NuGet.Clients/VsExtension/VisualStudioCredentialProvider.cs

```diff
@@ -89,7 +89,7 @@ public VisualStudioCredentialProvider(IVsWebProxy webProxyService)
                 // or request credentials. 
                 WebRequest.DefaultWebProxy = new WebProxy(uriToDisplay);
 
-                return PromptForCredentials(uri, cancellationToken);
+                return Task.FromResult(PromptForCredentials(uri, cancellationToken));
```

Comment:
it calls blocking `Shell.ThreadHelper.JoinableTaskFactory.Run`. So if it can be replaced with its async counterpart then the whole call chain would be async.